src/logic/safe/transactions/gas.ts

```diff
@@ -296,28 +296,34 @@ export const estimateGasForTransactionExecution = async ({
 }: TransactionExecutionEstimationProps): Promise<number> => {
   const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
   try {
-    if (approvalAndExecution) {
-      console.info(`Estimating transaction success for execution & approval...`)
+    let gasEstimation
+    // If safeTxGas === 0 we still have to estimate the gas limit to execute the transaction so we need to get an estimation
+    if (approvalAndExecution || safeTxGas === 0) {
```

Comment:
why is `|| safeTxGas === 0` needed? 

shouldn't this enter the `ìf`no matter the safeTxGas value?